src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeConstructorInfo.CoreCLR.cs

```diff
@@ -118,6 +119,15 @@ public override string ToString()
 
             return m_toString;
         }
+
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
```

Comment:
nit: for clarity instead of `==` which can be overloaded, use `object.ReferenceEquals(this, obj)`